ticket-management/ticket-management/Controllers/EventController.cs

```diff
@@ -1,99 +1,82 @@
 
-            /*
-            var dtoEvents = events.Select(e => new EventDTO()
-            {
-                EventDescription = e.EventDescription,
-                EventName = e.EventName,
-                EventType = e.EventType.EventTypeName,
-                Venue = e.Venue?.VenueLocation
-            });
-            */
-
-            var dtoEvents = _mapper.Map<List<EventDTO>>(events);
-
-            return Ok(dtoEvents);
+            return Ok(events);
         }
 
-
         [HttpGet]
-        public async Task<ActionResult<EventDTO>> GetById(long id)
+        public async Task<ActionResult<List<EventDTO>>> GetEvents([FromQuery] long? venueId, [FromQuery] string? eventTypeName)
         {
-            var @event = await _eventRepository.GetById(id);
-
-            if (@event == null)
+            if (venueId == null && eventTypeName == null)
             {
-                return NotFound();
+                var events = await _eventService.GetAll();
+                return Ok(events);
             }
-
-            /*
-            var dtoEvent = new EventDTO()
+            else if (venueId != null && eventTypeName == null)
             {
-                EventDescription = @event.EventDescription,
-                EventName = @event.EventName,
-                EventType = @event.EventType.EventTypeName,
-                Venue = @event.Venue?.VenueLocation
-            };
-            */
-
-            var dtoEvent = _mapper.Map<EventDTO>(@event);
-
-            return Ok(dtoEvent);
+                var events = await _eventService.GetAllByVenue(venueId.Value);
+                return Ok(events);
+            }
+            else if (venueId == null && eventTypeName != null)
+            {
+                var events = await _eventService.GetAllByType(eventTypeName);
+                return Ok(events);
+            }
+            else if (venueId != null && eventTypeName != null)
+            {
+                var events = await _eventService.GetAllByVenueAndType(venueId.Value, eventTypeName);
+                return Ok(events);
+            }
+            else
+            {
+                var events = new List<EventDTO>();
+                return Ok(events);
+            }
         }
 
+
         [HttpPatch]
         public async Task<ActionResult<EventPatchDTO>> Patch(EventPatchDTO eventPatch)
         {
-            var eventEntity = await _eventRepository.GetById(eventPatch.EventId);
+            var result = await _eventService.Update(eventPatch);
 
-            if (eventEntity == null)
+            if (!result)
             {
                 return NotFound();
             }
 
-            if (!string.IsNullOrEmpty(eventPatch.EventName)) eventEntity.EventName = eventPatch.EventName;
-            if (!string.IsNullOrEmpty(eventPatch.EventDescription)) eventEntity.EventDescription = eventPatch.EventDescription;
-
-            _eventRepository.Update(eventEntity);
-
             return NoContent();
```

Comment:
returnezi doar no content sau not found? si daca totul e bine? :)

Reply:
Mă gândeam să returnez NoContent() dacă totul este okay și NotFound() dacă entitatea nu este găsită. Nu este în regulă?

Reply:
HTTP response status code 204 No Content is returned by the server to indicate that a HTTP request has been successfully completed, and there is no message body. Adica, pe scurt, cineva face un request catre api tau dar fara actual content (request gol). De ex sa iti zica cineva sa stergi ceva, dar fara sa iti specifice exact ce sa stearga. Eu cel mai des am vazut de genul ca daca s-a sters cu success, se returneaza ok catre requester, iar el daca primeste acel ok afiseaza un mesaj de genul "sters cu success'.